tests/test_specialize.py

```diff
@@ -91,6 +92,14 @@ def run_test(args):
                     raise verr
 
                 result_final = res['output'](*args)
+
+                if isinstance(result_final, ArrayWrapper):
+                    result_final = result_final.array
+                elif isinstance(result_final, tuple):
+                    result_final = tuple(a.array if isinstance(a, ArrayWrapper) else a for a in result_final)
+                elif isinstance(result_final, list):
+                    result_final = [a.array if isinstance(a, ArrayWrapper) else a for a in result_final]
```

Comment:
This should be unnecessary, because `return_backend` is False (see my other comment about this).